internal/terraform/eval_variable.go

```diff
@@ -140,6 +132,14 @@ func prepareFinalInputVariableValue(addr addrs.AbsInputVariableInstance, raw *In
 		return cty.UnknownVal(cfg.Type), diags
 	}
 
+	// Apply defaults from the variable's type constraint to the given value,
+	// unless the given value is null. We do not apply defaults to top-level
```

Comment:
```suggestion
	// Apply defaults from the variable's type constraint to the converted value,
	// unless the converted value is null. We do not apply defaults to top-level
```

The use of the word "given" here was intentionally referring to the `given` variable, which is unconverted. The suggestion in this comment is one way to address the issue I have with this comment, which is that I think applying defaults after conversion is a bit subtle and we should try to leave a note about it.